packages/modelence/src/migration/index.ts

```diff
@@ -1,35 +1,98 @@
+import { acquireLock, releaseLock } from '@/lock';
 import { Module } from '../app/module';
 import { dbMigrations } from './db';
+import { logInfo } from '../telemetry';
 
 export type MigrationScript = {
   version: number;
   description: string;
-  handler: () => Promise<void>;
+  handler: () => Promise<string | void>;
 };
 
 export async function runMigrations(migrations: MigrationScript[]) {
   if (migrations.length === 0) {
     return;
   }
 
+  const hasLock = await acquireLock('migrations');
```

Comment:
What happens if the current owner of the lock is a different version of the app that doesn't have all the up to date migrations?

More of a theoretical question since I find it hard to imagine how that would happen, but still would be good to have some guardrails for that, like attempting to run migrations periodically while there are pending scripts and ending that cycle only when there are no more pending scripts.